util/caller/resolver.go

```diff
@@ -45,8 +45,20 @@ type CallResolver struct {
 }
 
 // defaultPattern strips src/github.com/organization/project/module/submodule/file.go
-// to module/submodule/file.go.
-var defaultPattern = "src/" + strings.Repeat(strings.Join([]string{"[^", "]+", ""}, string(os.PathSeparator)), 3) + "(.*)"
+// down to module/submodule/file.go.
+var defaultPattern = func() string {
+	_, file, _, ok := runtime.Caller(0)
+	if !ok {
+		panic("unable to look up location")
+	}
+	const sep = string(os.PathSeparator)
+	path := filepath.Dir(file)
+	// Strip to $GOPATH/src.
+	for i := 0; i < 5; i++ {
+		path = filepath.Dir(filepath.Clean(path))
+	}
+	return path + sep + strings.Repeat(strings.Join([]string{"[^", "]+", ""}, sep), 3) + "(.*)"
```

Comment:
Backslashes are special characters in regular expressions; we need to use `regexp.QuoteMeta` each time we refer to `path` and `sep` in this line. 

We might as well run `regexp.MustCompile` here instead of compiling it every time we create a CallResolver.